ksqldb-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4

```diff
@@ -333,7 +333,7 @@ type
     ;
 
 typeParameter
-    : INTEGER_VALUE | type
+    : INTEGER_VALUE | STRING
```

Comment:
I don't think the `STRING` token works here. See below:
```
STRING
    : '\'' ( ~'\'' | '\'\'' )* '\''
    ;
```

The `STRING` token will translate to any string that starts/ends with single quotes. So the `VARCHAR(STRING)` will fail. Could you add a test that verifies `VARCHAR(STRING)` works? I'm pretty sure that will fail, so you'll have to update the syntax.

Reply:
Yup, You’re right... I misread the first comment. 
It works with `'STRING'`, however breaks the syntax  with `STRING`.
<img width="468" alt="Screenshot 2021-06-10 at 08 11 16" src="https://user-images.githubusercontent.com/10224618/121481728-56aec780-c9c4-11eb-99c9-02e4f5c35d8b.png">

To solve it, `'STRING'`  has also to be added in `nonReserved`.

Reply:
@spena take it a look when you got a chance, thanks!